File: TCP  IP Server Client Example Part 6 (Chat Application)/[C#]-TCP  IP Server Client Example Part 6 (Chat Application)/C#/ChatClient/FrmLogin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ChatClient
{
    public partial class FrmLogin : Form
    {
        public bool IsRegistered { get; set; }
        public string Password { get; set; }
        public string EmailAddress { get; set; }

        public FrmLogin()
        {
            InitializeComponent();
            this.IsRegistered = false; // This is changed if the EU clicks the Register link and successfully registers
        }

        /// <summary>
        /// Cancel the login process
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.Close();
        }

        /// <summary>
        /// Add your own validation just as it was done in the AzureDB Login Form
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnLogin_Click(object sender, EventArgs e)
        {
            if (!string.IsNullOrWhiteSpace(this.tbEmailAddress.Text) && !string.IsNullOrWhiteSpace(this.tbPassword.Text))
            {

                this.EmailAddress = tbEmailAddress.Text;
                this.Password = tbPassword.Text.Replace(",", "");

                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();

            }
            else
            {
                MessageBox.Show("Login details are in valid");
            }
        }

        /// <summary>
        /// Let the EU Register from here.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void llRegister_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            FrmRegister register = new FrmRegister();
            DialogResult dr = new DialogResult();
            dr = register.ShowDialog();

            if (dr == System.Windows.Forms.DialogResult.OK)
            {
                this.EmailAddress = register.RegisteredUserEmail;
                this.Password = register.RegisteredUserPassword.Replace(",", "");
                this.IsRegistered = true;
                this.Close();
            }
            else
            {
                this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
                this.Close();
            }
        }
    }
}

File: TCP  IP Server Client Example Part 7 (Chat Application)/[VB]-TCP  IP Server Client Example Part 7 (Chat Application)/VB/ChatClient/FrmLogin.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ChatClient
	Partial Public Class FrmLogin
		Inherits Form

		Public Property IsRegistered() As Boolean
		Public Property Password() As String
		Public Property EmailAddress() As String

		Public Sub New()
			InitializeComponent()
			Me.IsRegistered = False ' This is changed if the EU clicks the Register link and successfully registers
		End Sub

		''' <summary>
		''' Cancel the login process
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCancel.Click
			Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
			Me.Close()
		End Sub

		''' <summary>
		''' Add your own validation just as it was done in the AzureDB Login Form
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnLogin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLogin.Click
			If (Not String.IsNullOrWhiteSpace(Me.tbEmailAddress.Text)) AndAlso (Not String.IsNullOrWhiteSpace(Me.tbPassword.Text)) Then

				Me.EmailAddress = tbEmailAddress.Text
				Me.Password = tbPassword.Text.Replace(",", "")

				Me.DialogResult = System.Windows.Forms.DialogResult.OK
				Me.Close()

			Else
				MessageBox.Show("Login details are in valid")
			End If
		End Sub

		''' <summary>
		''' Let the EU Register from here.
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub llRegister_LinkClicked(ByVal sender As Object, ByVal e As LinkLabelLinkClickedEventArgs) Handles llRegister.LinkClicked
			Dim register As New FrmRegister()
			Dim dr As New DialogResult()
			dr = register.ShowDialog()

			If dr = System.Windows.Forms.DialogResult.OK Then
				Me.EmailAddress = register.RegisteredUserEmail
				Me.Password = register.RegisteredUserPassword.Replace(",", "")
				Me.IsRegistered = True
				Me.Close()
			Else
				Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
				Me.Close()
			End If
		End Sub
	End Class
End Namespace
